Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -788,40 +788,44 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
-    if let unexpected = node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers,
-      let name = unexpected.presentTokens(satisfying: { $0.tokenKind.isIdentifier == true }).only
-    {
-      addDiagnostic(
-        name,
-        .deinitCannotHaveName,
-        fixIts: [
-          FixIt(message: RemoveNodesFixIt(name), changes: .makeMissing(name))
-        ],
-        handledNodes: [name.id]
-      )
+
+    var name: TokenSyntax? {
+      return node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers?.presentTokens(satisfying: { $0.tokenKind.isIdentifier == true }).only
     }
-    if let unexpected = node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers,
-      let params = unexpected.compactMap({ $0.as(FunctionParameterClauseSyntax.self) }).only
-    {
-      addDiagnostic(
-        params,
-        .deinitCannotHaveParameters,
-        fixIts: [
-          FixIt(message: RemoveNodesFixIt(params), changes: .makeMissing(params))
-        ],
-        handledNodes: [params.id]
-      )
+
+    var params: FunctionParameterClauseSyntax? {
+      return node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers?.compactMap({ $0.as(FunctionParameterClauseSyntax.self) }).only
     }
-    if let unexpected = node.unexpectedBetweenEffectSpecifiersAndBody,
-      let returnType = unexpected.compactMap({ $0.as(ReturnClauseSyntax.self) }).only
-    {
+
+    var returnType: ReturnClauseSyntax? {
+      return node.unexpectedBetweenEffectSpecifiersAndBody?.compactMap({ $0.as(ReturnClauseSyntax.self) }).only
+    }
+
+    var nodes: [any SyntaxProtocol] = []
+
+    if let name {
+      nodes.append(name)
+    }
+
+    if let params {
+      nodes.append(params)
+    }
+
+    if let returnType {
+      nodes.append(returnType)
+    }
```

Comment:
I wasn't able to get it compile by using `let nodes: [any Syntaxprotocol] = [name, params, returnType].compactMap { $0 }`

Reply:
It appears that you need to cast the default array to make it work. You can do this in one of the following ways:

Option 1:
```swift
let nodes: [AnySyntaxProtocol] = ([name, params, returnType] as [AnySyntaxProtocol?]).compactMap { $0 }
```

Option 2:
```swift
let optionalNodes: [AnySyntaxProtocol?] = [name, params, returnType]
let nodes: [AnySyntaxProtocol] = optionalNodes.compactMap { $0 }
```

Reply:
Hmm, I can't find the `AnySyntaxProtocol` type?
Is that something that isn't on main yet?

Reply:
Sorry, I'm using Grammarly to correct my message and sometimes it changes code blocks... I've edited my comment

Reply:
Thanks!

Reply:
What about 

```swift
    let name: TokenSyntax? = node
      .unexpectedBetweenDeinitKeywordAndEffectSpecifiers?
      .presentTokens(satisfying: { $0.tokenKind.isIdentifier == true })
      .only

    let params: FunctionParameterClauseSyntax? = node
      .unexpectedBetweenDeinitKeywordAndEffectSpecifiers?
      .compactMap({ $0.as(FunctionParameterClauseSyntax.self) })
      .only

    let returnType: ReturnClauseSyntax? = node
      .unexpectedBetweenEffectSpecifiersAndBody?
      .compactMap({ $0.as(ReturnClauseSyntax.self) })
      .only

    let nodes = [Syntax(name), Syntax(params), Syntax(returnType)].compactMap { $0 }
```

Reply:
Nice 😍